cmd/kubeless/function.go

```diff
@@ -38,3 +45,71 @@ func init() {
 	functionCmd.AddCommand(describeCmd)
 	functionCmd.AddCommand(updateCmd)
 }
+
+func constructFunction(runtime, handler, file, funcName, funcType, topic, ns, deps, description string, mem resource.Quantity, labels, envs []string) *spec.Function {
+	funcLabels := map[string]string{}
+	for _, label := range labels {
+		k, v := getKV(label)
+		funcLabels[k] = v
+	}
+
+	funcEnv := map[string]string{}
+	for _, env := range envs {
+		k, v := getKV(env)
+		funcEnv[k] = v
+	}
+
+	funcAnnotation := map[string]string{
+		"Description": description,
+	}
+
+	f := &spec.Function{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Function",
+			APIVersion: "k8s.io/v1",
+		},
+		Metadata: metav1.ObjectMeta{
+			Name:        funcName,
+			Namespace:   ns,
+			Labels:      funcLabels,
+			Annotations: funcAnnotation,
+		},
+		Spec: spec.FunctionSpec{
+			Handler:  handler,
+			Runtime:  runtime,
+			Type:     funcType,
+			Function: readFile(file),
+			Topic:    topic,
+			Env:      funcEnv,
+			Memory:   mem,
+		},
+	}
+	// add dependencies file to func spec
+	if deps != "" {
+		f.Spec.Deps = readFile(deps)
+	}
+
+	return f
+}
+
+func getKV(input string) (string, string) {
+	var key, value string
+	if pos := strings.IndexAny(input, "=:"); pos != -1 {
+		key = input[:pos]
+		value = input[pos+1:]
+	} else {
+		// no separator found
+		key = input
+		value = ""
+	}
+
+	return key, value
+}
+
+func readFile(file string) string {
+	data, err := ioutil.ReadFile(file)
+	if err != nil {
+		logrus.Fatalf("Can not read file: %s. The file may not exist", file)
```

Comment:
.. but then it will go on to create a Function with the content missing. How is that ok?